extensions/wikia/FlowTracking/scripts/createNewPageFlows.js

```diff
@@ -6,23 +6,31 @@ define(
 			articleId = mw.config.get('wgArticleId');
 
 		function trackOnEditPageLoad(editor) {
+			var qs = new QueryString(),
+				// 'flow' is the parameter passed in the url if user has started a flow already
+				flowParam = qs.getVal('flow', false);
+
 			// Track only creating articles (wgArticleId=0) from namespace 0 (Main)
 			// IMPORTANT: on Special:CreatePage even after providing article title the namespace is set to -1 (Special Page)
 			if (namespaceId === 0 && articleId === 0) {
-				var qs = new QueryString(),
-					// 'flow' is the parameter passed in the url if user has started a flow already
-					flowParam = qs.getVal('flow', false);
-
 				if (flowParam || document.referrer) {
-					//TODO: track middle step for other flows
+					flowTrack.trackFlowStep(flowParam, {editor: editor});
 				} else {
 					flowTrack.beginFlow(flowTrack.flows.CREATE_PAGE_DIRECT_URL, {editor: editor});
 				}
 			}
+
+			// For Special:CreatePage
+			if (namespaceId === -1 && articleId === 0) {
+				if (flowParam || document.referrer) {
+					flowTrack.trackFlowStep(flowParam, {editor: editor});
+				} else {
+					// TODO: direct-url to Special:CreatePage (WW-351)
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS1135)